crates/sui-core/src/authority/submitted_transaction_cache.rs

```diff
@@ -0,0 +1,343 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use consensus_types::block::Round;
+use parking_lot::RwLock;
+use std::collections::{BTreeMap, HashMap, HashSet};
+use std::net::IpAddr;
+use sui_types::digests::TransactionDigest;
+use sui_types::traffic_control::Weight;
+use tracing::debug;
+
+/// The number of consensus rounds to retain transaction submission information before garbage collection.
+pub(crate) const SUBMISSION_RETENTION_ROUNDS: u32 = 400;
```

Comment:
Since this records own submission only, maybe use a higher round like 1000? Otherwise `CONSENSUS_STATUS_RETENTION_ROUNDS` can be reused.
Alternatively, LRU seems good enough for this use case, instead of GC by round.